include/javabind/chrono.hpp

```diff
@@ -0,0 +1,123 @@
+
+    template <>
+    struct JavaDurationTraits<std::chrono::nanoseconds>
+    {
+        constexpr static std::string_view to_native_method = "toNanos";
+        constexpr static std::string_view of_native_method = "ofNanos";
+    };
+
+    template <>
+    struct JavaDurationTraits<std::chrono::milliseconds>
+    {
+        constexpr static std::string_view to_native_method = "toMillis";
+        constexpr static std::string_view of_native_method = "ofMillis";
+    };
+
+    template <>
+    struct JavaDurationTraits<std::chrono::seconds>
+    {
+        constexpr static std::string_view to_native_method = "getSeconds";
+        constexpr static std::string_view of_native_method = "ofSeconds";
+    };
+
+    template <>
+    struct JavaDurationTraits<std::chrono::minutes>
+    {
+        constexpr static std::string_view to_native_method = "toMinutes";
+        constexpr static std::string_view of_native_method = "ofMinutes";
+    };
+
+    template <>
+    struct JavaDurationTraits<std::chrono::hours>
+    {
+        constexpr static std::string_view to_native_method = "toHours";
+        constexpr static std::string_view of_native_method = "ofHours";
+    };
+
+    template <typename T>
+    struct JavaDurationType : public AssignableJavaType<T>
+    {
+        using native_type = T;
+        using java_type = jobject;
+
+        static native_type native_value(JNIEnv* env, java_type javaValue)
+        {
+            LocalClassRef durationClass(env, javaValue);
+            auto toNative = durationClass.getMethod(JavaDurationTraits<native_type>::to_native_method, "()J");
+            return native_type { env->CallLongMethod(javaValue, toNative.ref()) };
+        }
+
+        static java_type java_value(JNIEnv* env, const native_type& nativeValue)
+        {
+            LocalClassRef durationClass(env, AssignableJavaType<T>::class_path);
+            auto ofNative = durationClass.getStaticMethod(JavaDurationTraits<native_type>::of_native_method, "(J)Ljava/time/Duration;");
+            return env->CallStaticObjectMethod(durationClass.ref(), ofNative.ref(), static_cast<jlong>(nativeValue.count()));
+        }
+    };
+
+    template <typename T>
+    struct JavaInstantType : public AssignableJavaType<T>
+    {
+        using native_type = T;
+        using java_type = jobject;
+
+        static native_type native_value(JNIEnv* env, java_type javaValue)
+        {
+            LocalClassRef instantClass(env, javaValue);
+            auto toMillis = instantClass.getMethod("toEpochMilli", "()J");
+            auto millis = env->CallLongMethod(javaValue, toMillis.ref());
+            return native_type { std::chrono::milliseconds { millis } };
+        }
+
+        static java_type java_value(JNIEnv* env, const native_type& nativeValue)
+        {
+            LocalClassRef instantClass(env, AssignableJavaType<T>::class_path);
+            auto ofMillis = instantClass.getStaticMethod("ofEpochMilli", "(J)Ljava/time/Instant;");
```

Comment:
I would suppose we may want to call the Java static method `Instant.ofEpochSecond(long epochSecond, long nanoAdjustment)`, which would allow us to pass the full precision of an `std::chrono::time_point`, including nanoseconds.

Reply:
Done